tests/app/celery/test_provider_tasks.py

```diff
@@ -492,3 +495,113 @@ def test_deliver_sms_non_retryables(
     notify_db_session.session.refresh(notification)
     assert notification.status == NOTIFICATION_PERMANENT_FAILURE
     assert notification.status_reason == status_reason
+
+
+def test_deliver_push_happy_path_icn(
+    client,
+    rmock,
+):
+    rmock.register_uri(
+        'POST',
+        f'{client.application.config["VETEXT_URL"]}/mobile/push/send',
+        json={'message': 'success'},
+        status_code=201,
+    )
+    formatted_payload = {
+        'appSid': DEAFULT_MOBILE_APP_TYPE,
+        'templateSid': '2222',
+        'icn': '3333',
+        'personalization': {'%MSG_ID': '4444'},
+    }
+
+    # Should run without exceptions
+    deliver_push(formatted_payload)
+
+
+def test_deliver_push_happy_path_topic(
+    client,
+    rmock,
+):
+    rmock.register_uri(
+        'POST',
+        f'{client.application.config["VETEXT_URL"]}/mobile/push/send',
+        json={'message': 'success'},
+        status_code=201,
+    )
+    formatted_payload = {
+        'appSid': DEAFULT_MOBILE_APP_TYPE,
+        'templateSid': '2222',
+        'topicSid': '3333',
+        'personalization': {'%MSG_ID': '4444'},
+    }
+
+    # Should run without exceptions
+    deliver_push(formatted_payload)
+
+
+@pytest.mark.parametrize(
+    'test_exception, status_code',
+    [
+        (ConnectTimeout(), None),
+        (HTTPError(response=Response()), 429),
+        (HTTPError(response=Response()), 500),
+        (HTTPError(response=Response()), 502),
+        (HTTPError(response=Response()), 503),
+        (HTTPError(response=Response()), 504),
+    ],
+)
+def test_deliver_push_retryable_exception(
+    client,
+    rmock,
+    test_exception,
+    status_code,
+):
+    if status_code is not None:
+        test_exception.response.status_code = status_code
+    rmock.register_uri(
+        'POST',
+        f'{client.application.config["VETEXT_URL"]}/mobile/push/send',
+        exc=test_exception,
+    )
+    formatted_payload = {
+        'appSid': DEAFULT_MOBILE_APP_TYPE,
+        'templateSid': '2222',
+        'icn': '3333',
+        'personalization': {'%MSG_ID': '4444'},
+    }
+
+    with pytest.raises(AutoRetryException):
+        deliver_push(formatted_payload)
+
+
+@pytest.mark.parametrize(
+    'test_exception, status_code',
+    [
+        (HTTPError(response=Response()), 400),
+        (HTTPError(response=Response()), 403),
+        (HTTPError(response=Response()), 405),
+        (RequestException(), None),
+    ],
+)
+def test_deliver_push_nonretryable_exception(
+    client,
+    test_exception,
+    status_code,
+    rmock,
+):
+    if status_code is not None:
+        test_exception.response.status_code = status_code
+    rmock.register_uri(
+        'POST',
+        f'{client.application.config["VETEXT_URL"]}/mobile/push/send',
+        exc=test_exception,
+    )
+    formatted_payload = {
+        'appSid': DEAFULT_MOBILE_APP_TYPE,
```

Comment:
see comment above about fixing this longstanding typo `DEAFULT_MOBILE_APP_TYPE`

Reply:
Updated.